### PR TITLE
gateway-api: Re-queue gateway for namespace change

### PR DESCRIPTION
**Pull request by sayboras**
### Description

This commit is to make sure that if there is any changes in namespace mentioned in gateway.spec.listener[].allowRoutes, the reconciliation loop should kick off for Gateway resource. For example, the namespace label can be added to be included or excluded from what mentioned in gateway.spec.listener[*].allowRoutes.

### CODE REVIEW

File: operator/pkg/gateway-api/controller.go [hunk 1]
```diff
@@ -188,6 +189,59 @@ func getGatewaysForSecret(ctx context.Context, c client.Client, obj client.Objec
 	return gateways
 }
 
+func getGatewaysForNamespace(ctx context.Context, c client.Client, ns client.Object) []types.NamespacedName {
```
Comment (mhofstetter) [inline operator/pkg/gateway-api/controller.go:192]: adding a unittest? these mapping functions are quite easy to test - even though one might need the fakek8sclient.

Comment (sayboras) [reply operator/pkg/gateway-api/controller.go:192]: good call out, I noticed that we didn't have unit tests for other similar function as well. Let me take this chance to improve converage in this area :medal_military:

Comment (sayboras) [reply operator/pkg/gateway-api/controller.go:192]: ah actually, I already have unit test for getGatewaysForSecrets :)

**Review by mhofstetter — APPROVED** [review]

**Review by youngnick — APPROVED** [review]

**Comment by sayboras** [comment]
/test-1.24-5.4

**Comment by sayboras** [comment]
/test-1.24-5.4

**Pull request merged**